cdc/processor/sinkmanager/manager.go

```diff
@@ -287,21 +282,47 @@ func (m *SinkManager) backgroundGC() {
 					zap.String("namespace", m.changefeedID.Namespace),
 					zap.String("changefeed", m.changefeedID.ID))
 				return
-			case gcEvent := <-m.engineGCChan:
-				if err := m.sourceManager.CleanByTable(gcEvent.tableID, gcEvent.cleanPos); err != nil {
-					log.Error("Failed to clean table in sort engine",
-						zap.String("namespace", m.changefeedID.Namespace),
-						zap.String("changefeed", m.changefeedID.ID),
-						zap.Int64("tableID", gcEvent.tableID),
-						zap.Error(err))
-					select {
-					case m.errChan <- err:
-					default:
-						log.Error("Failed to send error to error channel, error channel is full",
+			case <-ticker.C:
+				tableSinks := make(map[model.TableID]*tableSinkWrapper)
+				m.tableSinks.Range(func(key, value any) bool {
+					tableID := key.(model.TableID)
+					wrapper := value.(*tableSinkWrapper)
+					tableSinks[tableID] = wrapper
+					return true
+				})
+
+				for tableID, sink := range tableSinks {
+					if time.Since(sink.lastCleanTime) < cleanTableInterval {
+						continue
+					}
+					checkpointTs := sink.getCheckpointTs()
+					resolvedMark := checkpointTs.ResolvedMark()
+					if resolvedMark == 0 {
+						continue
+					}
+					cleanPos := engine.Position{StartTs: resolvedMark - 1, CommitTs: resolvedMark}
+					if err := m.sourceManager.CleanByTable(tableID, cleanPos); err != nil {
+						log.Error("Failed to clean table in sort engine",
 							zap.String("namespace", m.changefeedID.Namespace),
 							zap.String("changefeed", m.changefeedID.ID),
+							zap.Int64("tableID", tableID),
 							zap.Error(err))
+						select {
+						case m.errChan <- err:
+						default:
+							log.Error("Failed to send error to error channel, error channel is full",
+								zap.String("namespace", m.changefeedID.Namespace),
+								zap.String("changefeed", m.changefeedID.ID),
+								zap.Error(err))
+						}
```

Comment:
What is the best behavior when error happens in the background goroutine
- to keep the goroutine running
- or let it return.

In other word, if the error is sent to the `errChan` or there exists an error in it already, the parent context will be canceled later, the opportunity of the logic in these background goroutines is indeterminate.